DoSomethingArchive/legacy-website
# Tests are great

## Changes:
- Moves [functional-testing](https://github.com/dosomething/functional-testing) into the main repo. This means we can build tests alongside features (a la TDD), and follows common practices for test organization in a project.
- Tests can be run using `ds test` or `ds test [directory or file]`. This will run CasperJS tests, and perform pre-test and post-test scripts. There's also a sneaky `CI_TESTING` argument to set the CasperJS `url` variable to the QA instance for running tests on QA.
- Some modifications to login, register, and campaign signup tests:
  - Using modal login/register form now to better reflect real user behavior.
  - Tests existence of proper headers on campaign signup test.
## To test this PR:

Re-provision your Vagrant box to ensure that PhantomJS and CasperJS are installed. Pull database from staging to ensure that you have the `QA_TEST_ACCOUNT@example.com` account.
## What to expect:

![screen shot 2014-07-03 at 5 13 12 pm](https://cloud.githubusercontent.com/assets/583202/3475987/f076bcec-02f7-11e4-9c46-d1ba30b3c9c8.png)
